### DISCUSSION TITLE
inlets / outlets through XCom API

### DISCUSSION THREAD

**Discussion by jdu** [apache/airflow#30587] [General]
Using the xcom API described here https://airflow.apache.org/docs/apache-airflow/stable/stable-rest-api-ref.html#operation/get_xcom_entry i'm trying to get the pipeline_inlets, and pipeline_outlets keys to process data about the inlets / outlets for specific tasks in our airflow deployment.

The value key for the xcom data are serialized objects, and because I want to run something fairly lightweight I can't see how to parse the values as they're not just standard old JSON.

Is there a way to deserialize the value for these xcom entries without having to install airflow itself?

**Comment by potiuk** [comment]
I believe you should rather look at the OpenLineage integration for Airflow, not try to parse XCom data, especially that we are incorporating Open lineage integratin to be part of Airflow https://cwiki.apache.org/confluence/display/AIRFLOW/AIP-53+OpenLineage+in+Airflow 

If what Open-Lineage integration is good for you, I'd strongly recommend on relying on what they do (you can even add your own extractors if you want to extract your own lineage data). 

Parsing XCom serialized form is brittle because it might (and lilely will) change - you could potentially look at teh source code of Airlfow and use the methods that are used to serialized/deserialize XCom data https://github.com/apache/airflow/blob/main/airflow/models/xcom.py#L619 - but we are (even as we speak) changing serialization mechanisms in Airlfow and there is no guarantee things will remain as they are in the future.

You are much better off (and much more future-proof) when you plug-in the lineage extraction by OpenLineage - this is much more robust way of getting inlet/outlet data obtaining (and with OL becoming de-facto standard, it's also nice way to make or plug-in whatever the lineage tool you use to understand Open Lineage  as there are also many other sources of OpenLineage data that you can plug-in your lineage tracking.
